### ISSUE TITLE
Sprite raster data missing from imageManager

### ISSUE THREAD

**Issue by ZeLonewolf** [maplibre/maplibre-gl-js#2159]
### Summary

OSM Americana relies on access to sprite raster data in map.style.imageManager.images in order to render highway shields in a "styleimagemissing" hook. At runtime, javascript code retrieves the raster data associated with a shield graphic, draws a route number on the graphic, and stores the new graphic back in the sprite sheet in memory. However, between 2.4.0 and the current 3.0.0 mainline code, sprite raster data is no longer accessible in imageManager.

If not corrected, OSM Americana would be unable to render highway shields after 2.4.0.

### Details

**maplibre-gl-js version**:

3.0.0-pre.3 (actually, current main branch as of the date of this bug report)

**browser**:

Google Chrome, Ubuntu

### Steps to Trigger Behavior

1. Any time after the map is loaded, run the code below.

```javascript
  for (const [key, value] of Object.entries(map.style.imageManager.images)) {
    console.log(`${key}, ${value.data}`);
  }
```
2. The expected output is a series of lines in the console that show:

```
<name of sprite>, [object Object]
```
3. However, in the current mainline, the output is:
```
<name of sprite>, null
```

### Expected Behavior

Clients are able to access sprite raster data in `map.style.imageManager.images`.

### Actual Behavior

Raster data is null in `map.style.imageManager.images`.

**Comment by HarelM** [comment]
I think this was optimized at some point in the near past.
You have the id, you should call `getImage` on that id in order to load the image into memory.
But generally speaking, I would recommend avoiding using internal structures such as those.
Why do you need to iterating over the images?

**Comment by ZeLonewolf** [comment]
The iterating was simply a bit of code to demonstrate the difference between the two versions.  I'll investigate getImage and hopefully that will be an easy fix.

**Comment by HarelM** [comment]
Sure, please close this issue if it was resolved.

**Comment by 1ec5** [comment]
[`listImages`](https://maplibre.org/maplibre-gl-js-docs/api/map/#map#listimages) should be an alternative for iterating over the image names.

**Comment by ZeLonewolf** [comment]
I was really just trying to see if 3.0.0 behaves differently from 2.4.0, which that bit of iteration code shows.

I intend to try converting our variable access to the getter version to see if the data is accessible there and if so whether that satisfies our need to access the sprite graphics at runtime.

**Comment by ZeLonewolf** [comment]
> I think this was optimized at some point in the near past. You have the id, you should call `getImage` on that id in order to load the image into memory. But generally speaking, I would recommend avoiding using internal structures such as those. Why do you need to iterating over the images?

I just spent some time looking through the [documentation](https://maplibre.org/maplibre-gl-js-docs/api/map/#map), however, it does not appear that there is a `getImage()` function.  Can you clarify where that's documented?

**Comment by HarelM** [comment]
In the code. Version 3 is still a pre-release and the docs are showing version 2.4.
I also review the PR that changed that behavior so this is partially from memory.
If you are using typescript with the latest version you should get the relevant auto complete, I hope.
In any case, if you think the documentation is lacking, I encourage you to open a PR. What is obvious or easy for me is probably not the case for others...

**Comment by ZeLonewolf** [comment]
Okay, looking through

https://github.com/maplibre/maplibre-gl-js/blob/main/src/ui/map.ts

It appears that map.style.getImage(id) is what I want.  I'm guessing there should probably be a map.getImage() that delegates to style.getImage() but not a roadblock.

**Comment by ZeLonewolf** [comment]
I attempted to implement the fix discussed here in ZeLonewolf/openstreetmap-americana#786. However, when switching from the internal variable to map.style.getImage(id), I discovered corrupted images in the map as shown in sample images #2165.

However, I've tested the approach described here locally with the fix in #2165, and with both changes combined it does appear to resolve the issue described in this ticket.
